### PR TITLE
*: update tools and fix lint

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

as title

### What is changed and how it works?

* update golangci-lint and many other tools version
* remove `golint` from golangci-lint config, because this linter is deprecated in new version

`WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.`
* fix some code style for new lint

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -11,7 +11,6 @@ linters:
     - exportloopref
     - gocritic
     - godot
-    - golint
```
Comment (GMHDBJD) [inline .golangci.yml:14]: Do we need to disable the whole golint linter？

Comment (GMHDBJD) [inline .golangci.yml:14]: LGTM. Maybe still remain it, though it's deprecated.

**Review by GMHDBJD — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

File: .golangci.yml (hunk 1 above)
Comment (Ehco1996) [reply .golangci.yml:14]: i think no need to keep `golint`, becasue it's check have been replaced by `revive` (we already have this linter)

**Pull request merged**